src/agent/border_agent.cpp

```diff
@@ -410,6 +428,28 @@ void BorderAgent::HandleExtPanId(void *aContext, int aEvent, va_list aArguments)
     static_cast<BorderAgent *>(aContext)->SetExtPanId(xpanid);
 }
 
+void BorderAgent::HandleThreadVersion(void *aContext, int aEvent, va_list aArguments)
+{
+    assert(aEvent == Ncp::kEventThreadVersion);
+
+    // `uint16_t` has been promoted to `int`.
+    uint16_t threadVersion = static_cast<uint16_t>(va_arg(aArguments, int));
+    static_cast<BorderAgent *>(aContext)->SetThreadVersion(threadVersion);
+}
+
+const char *BorderAgent::ThreadVersionToString(uint16_t aThreadVersion)
```

Comment:
I don't think this is necessary to be member method of BorderAgent. How about make this a local function?

Reply:
this is a stateless static member function.

Reply:
use static local function, we reduce a symbol exposed to outside the class. And I think it can be compiled faster.